src/libAtomVM/otp_socket.c

```diff
@@ -163,24 +164,23 @@ struct SocketResource
         struct ListHead udp_received_list;
         struct SocketResource *selected_client;
     };
+    size_t buf_size;
```

Comment:
This probably is unused.

Reply:
Actually there was a compilation error without it, but that might have been because the code that manipulates the buffer size is outside of the #ifdefs

Reply:
Line 769 I guess.

Reply:
Same comment as above -- since this is at the OTP level, can we support this in Pico too?

Reply:
If the semantics is about limiting the number of bytes to this value if passed length is 0, could we do it in this PR?